### ISSUE TITLE
[typespec-next] Swagger BreakingChange error "AddedAdditionalProperties"

### ISSUE THREAD

**Issue by mikeharder** [Azure/azure-rest-api-specs#25382]
The swagger updates caused by https://github.com/Azure/typespec-azure/pull/3391 raised these errors in check `Swagger BreakingChange`:

Rule | Message
-- | --
❌ 1021 - AddedAdditionalProperties | The new version adds an 'additionalProperties' element.<br>New: inference/preview/2023-06-01-preview/generated.json#L794:11<br>Old: inference/preview/2023-06-01-preview/generated.json#L794:11
❌ 1021 - AddedAdditionalProperties | The new version adds an 'additionalProperties' element.<br>New: inference/preview/2023-07-01-preview/generated.json#L827:11<br>Old: inference/preview/2023-07-01-preview/generated.json#L827:11
❌ 1021 - AddedAdditionalProperties | The new version adds an 'additionalProperties' element.<br>New: inference/stable/2022-12-01/generated.json#L351:11<br>Old: inference/stable/2022-12-01/generated.json#L351:11
❌ 1021 - AddedAdditionalProperties | The new version adds an 'additionalProperties' element.<br>New: inference/stable/2023-05-15/generated.json#L592:11<br>Old: inference/stable/2023-05-15/generated.json#L592:11

https://github.com/Azure/azure-rest-api-specs/pull/25323/checks?check_run_id=15930722524

@timotheeguerin: Can you comment on this?  If this is a valid breaking change to the swagger, what are the implications for merging this to main at the next TypeSpec release?

**Comment by timotheeguerin** [comment]
Yeah that would technically be a breaking change if that swagger was used to generated as it would go from being an array of `unknown[]` to `(float32 | null)[]` but because this is data plane then it is only for reference so shouldn't have implications
